### PR TITLE
Make Custom Elements v1 work with Safari 9, and add cross-browser tests

### PR DESCRIPTION
**Pull request by tdumitrescu**
Travis integration coming

I figure since we rely on this lib pretty fundamentally for resolving some browser- and transpilation-specific issues, we should be running tests across a decent range of browsers.

Tests exercise:
- WebComponent convenience methods (`getJSONAttribute`, `isAttributeEnabled`)
- basic Custom Elements methods; this is mainly to ensure that the shims/polyfills for Safari etc don't break basic functionality

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,8 +5,12 @@
   "main": "build/index.js",
   "scripts": {
     "build": "babel lib -d build",
+    "build-test": "webpack --config test/webpack.config.js",
     "prepublish": "npm run build",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "npm run build-test && npm run test-local",
```
Comment (nojvek) [inline package.json]: Niiiice!

File: test/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const path = require(`path`);
+
+const webpackConfig = {
```
Comment (nojvek) [inline test/webpack.config.js]: not sure if you care about verbosity of webpack but I always find it annoying that it pastes a huge chunk.

I like to add

```
  stats: {
    modules: false,
    children: false,
    chunks: false,
    assets: true,
    assetsSort: `name`,
    timings: true,
  },
```

File: demo/webpack.config.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ var webpackConfig = {
         exclude: /node_modules/,
         loader: 'babel',
         query: {
-          presets: ['es2015'],
+          presets: ['env'],
```
Comment (nojvek) [inline demo/webpack.config.js]: what does `env` do?

Comment (nojvek) [reply demo/webpack.config.js]: Ahh it's basically the shiniest babel. Nice

File: test/webpack.config.js (hunk 2 above)
Comment (tdumitrescu) [reply test/webpack.config.js]: aight https://github.com/mixpanel/webcomponent/pull/5/commits/8b10196415b5d30e7f7d1f53c9faebd9dede1759

File: demo/webpack.config.js (hunk 3 above)
Comment (tdumitrescu) [reply demo/webpack.config.js]: Yeah, for right now we'll get all of ES6-8 with this. But the real value is that then we can start specifying minimum browser versions we want to support via [browserslist](https://github.com/ai/browserslist) and Babel will only transpile what it needs to for our target envs.

**Pull request merged**